### PR TITLE
Extending the Unicode Emoji based on the 'unicode_smiles addon' 

### PR DESCRIPTION
**Pull request by loma-one**
Extending the Unicode Emoji based on the 'unicode_smiles addon'

### CODE REVIEW

File: unicode_smilies/unicode_smilies.php [hunk 1]
```diff
@@ -1,260 +1,2042 @@
...
-	Smilies::add($b, ':darts', '&#x1F3AF;');
-	Smilies::add($b, ':fencing', '&#x1F93A;');
-	Smilies::add($b, ':juggling', '&#x1F939;');
-//	Smilies::add($b, ':skipping', '&#x;');
-//	Smilies::add($b, ':archery', '&#x;');
-	Smilies::add($b, ':surfing', '&#x1F3C4;');
-	Smilies::add($b, ':snooker', '&#x1F3B1;');
-	Smilies::add($b, ':horseriding', '&#x1F3C7;');
+Smilies::add($b, ':basketball', '&#x1F3C0;');
+Smilies::add($b, '~bowling', '&#x1F3B3;');
```
Comment (tobiasd) [inline unicode_smilies/unicode_smilies.php]: This should be `:bowling` and not `-bowling` I think.

File: unicode_smilies/unicode_smilies.php [hunk 2]
```diff
@@ -1,260 +1,2042 @@
...
+Smilies::add($b, ':apple', '&#x1F34E;');
+//Smilies::add($b, ':broccoli', '&#x;');
+Smilies::add($b, ':cake', '&#x1F370;');
+//Smilies::add($b, ':carrot', '&#x1F955;'); // Bad client support
+Smilies::add($b, ':popcorn', '&#x1F37F;');
+Smilies::add($b, ':tomato', '&#x1F345;');
+Smilies::add($b, ':banana', '&#x1F34C;');
+Smilies::add($b, ':cooking', '&#x1F373;');
+Smilies::add($b, ':fryegg', '&#x1F373;');
+Smilies::add($b, ':birthday cake', '&#x1F382;');
```
Comment (tobiasd) [inline unicode_smilies/unicode_smilies.php]: Do spaces within the smiley definition work?

File: unicode_smilies/unicode_smilies.php [hunk 3]
```diff
@@ -1,260 +1,2042 @@
...
+    Smilies::add($b, ':grinsendes schiel-gesicht', '&#x1F606');
+    Smilies::add($b, ':lachen', '&#x1F606');
+    Smilies::add($b, ':grinsendes gesicht schweißgebadetes', '&#x1F605');
+    Smilies::add($b, ':sich vor Lachen auf dem boden wälzend', '&#x1F923');
+    Smilies::add($b, ':gesicht vor freudentränen weinen', '&#x1F602');
+    Smilies::add($b, ':leichtes lachen', '&#x1F642');
+    Smilies::add($b, ':verkehrtes gesicht', '&#x1F643');
+    Smilies::add($b, ':zwinkerndes gesicht', '&#x1F609');
+    Smilies::add($b, ':lachendes gesicht mit lächelnden augen', '&#x1F60A');
+    Smilies::add($b, ':lachendes gesicht mit heiligenschein', '&#x1F607');
```
Comment (tobiasd) [inline unicode_smilies/unicode_smilies.php]: I'm unsure about these (few) German translations for the emoticons. Why those and not all, and why not every language that exists on Earth? ;-)

File: unicode_smilies/unicode_smilies.php [hunk 4]
```diff
@@ -1,260 +1,2042 @@
 <?php
 /*
  * Name: Unicode Smilies
- * Description: Smilies based on the unicode emojis - On Linux use https://github.com/eosrei/emojione-color-font to see them in color
+ * Description: Smilies based on the unicode emojis - On Linux use https://github.com/eosrei/emojione-color-font to see them in color and http://www.unicode.org/emoji/charts/full-emoji-list.html
  * Version: 1.0
  * Author: Michael Vogel <http://pirati.ca/profile/heluecht>
```
Comment (tobiasd) [inline unicode_smilies/unicode_smilies.php]: I think your expansion is worth a higher version number, and maybe a mention in the Author list (just add another `* Author: ...` line if you like),

File: unicode_smilies/unicode_smilies.php [hunk 5]
```diff
@@ -1,260 +1,2042 @@
...
 
 function unicode_smilies_install() {
-	Hook::register('smilie', 'addon/unicode_smilies/unicode_smilies.php', 'unicode_smilies_smilies');
+Hook::register('smilie', 'addon/unicode_smilies/unicode_smilies.php', 'unicode_smilies_smilies');
 }
 
 function unicode_smilies_uninstall() {
-	Hook::unregister('smilie', 'addon/unicode_smilies/unicode_smilies.php', 'unicode_smilies_smilies');
+Hook::unregister('smilie', 'addon/unicode_smilies/unicode_smilies.php', 'unicode_smilies_smilies');
 }
```
Comment (tobiasd) [inline unicode_smilies/unicode_smilies.php]: I think you can remove the `uninstall` function. Right @MrPetovan ?

**Comment by loma-one** [comment]
The changes have been made. I hope this fits.
The changes are probably available as "loma-one:patch-1". I don't know if that's right.

**Comment by annando** [comment]
The indentation should be a single tab. Can you also have a look at the things that @tobiasd pointed out?

**Comment by loma-one** [comment]
Die Hinweise von @tobiasd wurden gleichfalls berücksichtigt.

File: unicode_smilies/unicode_smilies.php (hunk 2 above)
Comment (loma-one) [reply unicode_smilies/unicode_smilies.php]: It works on my instance

File: unicode_smilies/unicode_smilies.php (hunk 3 above)
Comment (loma-one) [reply unicode_smilies/unicode_smilies.php]: They are removing

File: unicode_smilies/unicode_smilies.php (hunk 4 above)
Comment (loma-one) [reply unicode_smilies/unicode_smilies.php]: okay

File: unicode_smilies/unicode_smilies.php (hunk 1 above)
Comment (loma-one) [reply unicode_smilies/unicode_smilies.php]: modified

File: unicode_smilies/unicode_smilies.php (hunk 5 above)
Comment (loma-one) [reply unicode_smilies/unicode_smilies.php]: I leave that to those who know about it ;)

Comment (MrPetovan) [reply unicode_smilies/unicode_smilies.php]: Absolutely, if it's only `Hook::unregister` calls.

File: unicode_smilies/unicode_smilies.php [hunk 6]
```diff
@@ -257,4 +264,1766 @@ function unicode_smilies_smilies(&$a,&$b) {
...
+	Smilies::add($b, ':white large square', '&#x2B1C');
+	Smilies::add($b, ':black medium square', '&#x25FC');
+	Smilies::add($b, ':white medium square', '&#x25FB');
+	Smilies::add($b, ':black medium-small square', '&#x25FE');
+	Smilies::add($b, ':white medium-small square', '&#x25FD');
+	Smilies::add($b, ':black small square', '&#x25AA');
+	Smilies::add($b, ':white small square', '&#x25AB');
+	Smilies::add($b, ':large orange diamond', '&#x1F536');
+	Smilies::add($b, ':large blue diamond', '&#x1F537');
+	Smilies::add($b, ':small orange diamond',
```
Comment (MrPetovan) [inline unicode_smilies/unicode_smilies.php]: Why didn't you use the semicolon syntax for this?

Comment (MrPetovan) [inline unicode_smilies/unicode_smilies.php]: Why does `wales` appear twice?

File: unicode_smilies/unicode_smilies.php [hunk 7]
```diff
@@ -257,4 +264,1766 @@ function unicode_smilies_smilies(&$a,&$b) {
...
+//	Smilies::add($b, ':brown circle', '&#x1F7E4');
+	Smilies::add($b, ':black circle', '&#x26AB');
+	Smilies::add($b, ':white circle', '&#x26AA');
+//	Smilies::add($b, ':red square', '&#x1F7E5');
+//	Smilies::add($b, ':orange square', '&#x1F7E7');
+//	Smilies::add($b, ':yellow square', '&#x1F7E8');
+//	Smilies::add($b, ':green square', '&#x1F7E9');
+//	Smilies::add($b, ':blue square', '&#x1F7E6');
+//	Smilies::add($b, ':purple square', '&#x1F7EA');
+//	Smilies::add($b, ':brown square', '&#x1F7EB');
```
Comment (MrPetovan) [inline unicode_smilies/unicode_smilies.php]: Why did you add commented out lines?

File: unicode_smilies/unicode_smilies.php (hunk 6 above)
Comment (loma-one) [reply unicode_smilies/unicode_smilies.php]: Since I'm not a developer and therefore I followed the guidelines of @annando  Programmatically, after I saw that it works ;)

File: unicode_smilies/unicode_smilies.php (hunk 7 above)
Comment (loma-one) [reply unicode_smilies/unicode_smilies.php]: They are defined in the standard system. So I adopted them. After I saw that they didn't display correctly, I commented them out.

File: unicode_smilies/unicode_smilies.php (hunk 6 above)
Comment (loma-one) [reply unicode_smilies/unicode_smilies.php]: ? I only see her once

Comment (MrPetovan) [reply unicode_smilies/unicode_smilies.php]: Ok, please use `:wales` instead.

Comment (MrPetovan) [reply unicode_smilies/unicode_smilies.php]: What about the duplicate `wales`? Can you please point me to the origin of the duplicate?

**Comment by MrPetovan** [comment]
Thanks for your patience!
